AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,1358 @@
+        }
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error in batch set parameters: " + str(e))
+        raise
+
+
+def delete_device(song, track_index, device_index, ctrl=None):
+    """Delete a device from a track."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)